sherpa/bin/conv_emformer_transducer_stateless2/beam_search.py

```diff
@@ -0,0 +1,254 @@
+        stream.hyp = []
+
+    @torch.no_grad()
+    def process(
+        self,
+        server: "StreamingServer",
+        stream_list: List[Stream],
+    ) -> None:
+        """Run the model on the given stream list and do search with given decoding
+           method.
+        Args:
+          server:
+            An instance of `StreamingServer`.
+          stream_list:
+            A list of streams to be processed. It is changed in-place.
+            That is, the attribute `states`, `decoder_out`, and `hyp` are
+            updated in-place.
+        """
+        model = server.model
+        device = model.device
+        # Note: chunk_length is in frames before subsampling
+        chunk_length = server.chunk_length
+        batch_size = len(stream_list)
+        chunk_length_pad = server.chunk_length_pad
+        state_list, feature_list = [], []
+        processed_frames_list, rnnt_decoding_streams_list = [], []
+
+        rnnt_decoding_config = self.rnnt_decoding_config
+        for s in stream_list:
+            rnnt_decoding_streams_list.append(s.rnnt_decoding_stream)
+            state_list.append(s.states)
+            processed_frames_list.append(s.processed_frames)
+            f = s.features[:chunk_length_pad]
+            s.features = s.features[chunk_length:]
+            s.processed_frames += chunk_length
+
+            b = torch.cat(f, dim=0)
+            feature_list.append(b)
+
+        features = torch.stack(feature_list, dim=0).to(device)
+
+        states = stack_states(state_list)
+
+        features_length = torch.full(
+            (batch_size,),
+            fill_value=features.size(1),
+            device=device,
+            dtype=torch.int64,
+        )
+
+        num_processed_frames = torch.tensor(
+            processed_frames_list, device=device
+        )
+
+        (
+            encoder_out,
+            encoder_out_lens,
+            next_states,
+        ) = model.encoder_streaming_forward(
+            features=features,
+            features_length=features_length,
+            num_processed_frames=num_processed_frames,
+            states=states,
+        )
+
+        processed_lens = (num_processed_frames >> 2) + encoder_out_lens
+        next_hyp_list = fast_beam_search_one_best(
+            model=model,
+            encoder_out=encoder_out,
+            processed_lens=processed_lens,
+            rnnt_decoding_config=rnnt_decoding_config,
+            rnnt_decoding_streams_list=rnnt_decoding_streams_list,
+        )
+
+        next_state_list = unstack_states(next_states)
+        for i, s in enumerate(stream_list):
+            s.states = next_state_list[i]
+            s.hyp = next_hyp_list[i]
+
+    def get_texts(self, stream):
+        """
+        Return text after decoding
+        Args:
+          stream:
+            Stream to be processed.
+        """
+        return self.sp.decode(stream.hyp[self.context_size :])
```

Comment:
Should it be
```suggestion
        return self.sp.decode(stream.hyp)
```
for `fast_beam_search`?

I think you only have to remove the leading `self.context_size` blanks for `modified_beam_search` and `greedy_search`.

Reply:
Indeed, if I am not mistaken I think it was not the case in the previous code

Reply:
>  I think it was not the case in the previous code

Yes, you are right. The previous code was wrong.

I just rechecked the code.

https://github.com/k2-fsa/sherpa/pull/68/files#diff-eb8856b1c35b482bd06279933a21bac2fce684949e1f8630faf1c51e01dd3e6e

![abc00](https://user-images.githubusercontent.com/5284924/180594788-67fcbe8f-1f77-403d-b6bc-5a98cf16408c.png)

`sherpa/bin/pruned_stateless_emformer_rnnt2/streaming_server.py` uses the correct version.

However, `sherpa/bin/conv_emformer_transducer_stateless/streaming_server.py` uses the incorrect version.